log-manager/src/main/java/io/airlift/log/LoggingOutputStream.java

```diff
@@ -44,7 +44,13 @@ public synchronized void flush()
         String record = this.toString();
         reset();
 
-        if (record.isEmpty() || record.equals(lineSeparator)) {
+        // Strip trailing new line typically added when this class is used via System.out.println
+        if (record.length() >= lineSeparator.length() &&
```

Comment:
I think we can eliminate `lineSeparator` and simplify this code to
```java
record = record.stripTrailing();
```

Reply:
pushed a change to strip all trailing new lines

the existing logic treats lineSeparator specially.
if we don't care about line endings, then yes, we can simplify this further

Reply:
I don’t think we care about platform line endings. Stripping all leading and trailing whitespace seems fine to me. Do you see any issues with that?

Reply:
i don't see a reason to strip leading whitespace and have no evidence yet that it would be useful.  it wasn't this code's behavior before my changes, so i didn't want to introduce it.

do you think that my changes require stripping leading whitespace as well?

Reply:
It can make stack traces look different to strip leading whitespace, but that doesn't bother me.